### PR TITLE
Document project_model::PackageData and project_model::TargetData

### PR DESCRIPTION
**Pull request by arnaudgolfouse**
This PR adds some documentation for the `project_model` crate.

Some of the field descriptions were taken directly from their `cargo_metadata` counterpart :
- `PackageData` -> `cargo_metadata::Package`
- `TargetData` -> `cargo_metadata::Target`

### CODE REVIEW

File: crates/project_model/src/cargo_workspace.rs [hunk 1]
```diff
@@ -80,19 +80,35 @@ pub type Package = Idx<PackageData>;
...
     pub name: String,
+    /// Path containing the `Cargo.toml`
     pub manifest: AbsPathBuf,
+    /// Targets provided by the crate (lib, bin, example, test, ...)
     pub targets: Vec<Target>,
+    /// Is this package a member of the current workspace
     pub is_member: bool,
+    /// List of packages this package depends on
     pub dependencies: Vec<PackageDependency>,
+    /// Default Rust edition for this package
```
Comment (matklad) [inline crates/project_model/src/cargo_workspace.rs:98]:
```suggestion
    /// Rust edition for this package
```

File: crates/project_model/src/cargo_workspace.rs [hunk 2]
```diff
@@ -80,19 +80,35 @@ pub type Package = Idx<PackageData>;
...
     pub edition: Edition,
+    /// List of features to activate
```
Comment (matklad) [inline crates/project_model/src/cargo_workspace.rs:100]:
```suggestion
    /// List of available features
```

**Review by matklad — COMMENTED** [review]
bors d+

File: crates/project_model/src/cargo_workspace.rs (hunk 2 above)
Comment (arnaudgolfouse) [reply crates/project_model/src/cargo_workspace.rs:100]: From what I could gather, this field is taken from the [Resolve](https://docs.rs/cargo_metadata/0.12.0/cargo_metadata/struct.Resolve.html) structure of `cargo metadata`, which contains [enabled](https://docs.rs/cargo_metadata/0.12.0/cargo_metadata/struct.Node.html#structfield.features) features.

Testing by hand also confirmed this

File: crates/project_model/src/cargo_workspace.rs (hunk 1 above)
Comment (arnaudgolfouse) [reply crates/project_model/src/cargo_workspace.rs]: Done :slightly_smiling_face:

**Pull request merged**